File: array/130.cpp
#include<bits/stdc++.h>

using namespace std;

class SeatManager {
public:
    SeatManager(int n) {
        for(int i = 1; i <= n ; ++i)
            seats.push(i);
    }
    
    int reserve() {
        int minSeat = seats.top();
        seats.pop();

        return minSeat;
    }
    
    void unreserve(int seatNumber) {
        seats.push(seatNumber);
    }

private:
    priority_queue<int, vector<int>, greater<int>> seats;    
    
};

int main() {
    SeatManager obj(10);
    cout<<obj.reserve();
    

    return 0;
}

File: array/130.py
from queue import PriorityQueue


class SeatManager:
    def __init__(self, n: int):
        self.pq = PriorityQueue()
        for item in range(1, n + 1):
            self.pq.put(item)

    def reserve(self) -> int:
        minSeat = self.pq.get()
        return minSeat

    def unreserve(self, seatNumber: int) -> None:
        self.pq.put(seatNumber)
